.Lib9c.Tests/Action/ClaimStakeRewardTest.cs

```diff
@@ -108,24 +108,57 @@ public void Serialization()
             Assert.Equal(action.AvatarAddress, deserialized.AvatarAddress);
         }
 
-        [Fact]
-        public void Execute_Success()
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void Execute_Success(bool useOldTable)
+        {
+            Execute(_avatarAddress, useOldTable);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void Execute_With_Old_AvatarState_Success(bool useOldTable)
         {
-            Execute(_avatarAddress);
+            Execute(_avatarAddressForBackwardCompatibility, useOldTable);
         }
 
         [Fact]
-        public void Execute_With_Old_AvatarState_Success()
+        public void Execute_Throw_ActionObsoletedException()
         {
-            Execute(_avatarAddressForBackwardCompatibility);
+            var action = new ClaimStakeReward(_avatarAddress);
+            Assert.Throws<ActionObsoletedException>(() => action.Execute(new ActionContext
+            {
+                PreviousStates = _initialState,
+                Signer = _signerAddress,
+                BlockIndex = ClaimStakeReward3.HardForkIndex + 1,
```

Comment:
If the `ClaimStakeReward3` is available in `ClaimStakeReward3.HardForkIndex`, and the `ClaimStakeReward`("xxx2") should throw an exception in `ClaimStakeReward3.HardForkIndex`. How you think about this?

```suggestion
                BlockIndex = ClaimStakeReward3.HardForkIndex,
```